modules/gapi/src/backends/oak/goakbackend.cpp

```diff
@@ -0,0 +1,597 @@
+    explicit GOAKKernel(const F& f) : m_f(f) {}
+    const F m_f;
+};
+
+struct OAKComponent
+{
+    static const char *name() { return "OAK Component"; }
+    GOAKKernel k;
+};
+
+}} // namespace gimpl // namespace cv
+
+using OAKGraph = ade::TypedGraph
+    < cv::gimpl::OAKComponent
+    // FIXME: extend
+    >;
+
+using ConstOAKGraph = ade::ConstTypedGraph
+    < cv::gimpl::OAKComponent
+    // FIXME: extend
+    >;
+
+void cv::gimpl::GOAKExecutable::LinkToParentHelper(ade::NodeHandle handle,
+                                                   const std::vector<ade::NodeHandle>& nodes)
+{
+    ade::NodeHandle parent;
+    for (const auto& indatah : handle.get()->inNodes()) {
+        // indatah - node's input data
+        // need to find which other node produces that data
+        for (const auto& nhp : nodes) {
+            if (m_gm.metadata(nhp).get<NodeType>().t == NodeType::OP) {
+                for (const auto& outdatah : nhp.get()->outNodes()) {
+                    if (indatah == outdatah) {
+                        parent = nhp;
+                    }
+                }
+            }
+        }
+    }
+    GAPI_Assert(m_oak_nodes.find(parent) != m_oak_nodes.end());
+    GAPI_Assert(m_oak_node_inputs[handle].size() ==
+                m_oak_node_outputs[parent].size());
+    for (size_t i = 0; i < m_oak_node_inputs[handle].size(); ++i) {
+        m_oak_node_outputs[parent][i]->link(*(m_oak_node_inputs[handle][i]));
+    }
+}
+
+cv::GArg
+cv::gimpl::GOAKExecutable::packArg(const GArg &arg,
+                                   std::vector<ExtractTypeHelper::InputPtr>& oak_ins,
+                                   std::vector<cv::GArg>& oak_wrapped_args) {
+    if (arg.kind != cv::detail::ArgKind::GOBJREF) {
+        GAPI_Assert(   arg.kind != cv::detail::ArgKind::GMAT
+                    && arg.kind != cv::detail::ArgKind::GSCALAR
+                    && arg.kind != cv::detail::ArgKind::GARRAY
+                    && arg.kind != cv::detail::ArgKind::GOPAQUE
+                    && arg.kind != cv::detail::ArgKind::GFRAME);
+        // All other cases - pass as-is, with no transformations to
+        // GArg contents.
+        return arg;
+    }
+    const cv::gimpl::RcDesc &ref = arg.get<cv::gimpl::RcDesc>();
+    switch (ref.shape) {
+    case GShape::GFRAME:
+        oak_ins.push_back(nullptr);
+        oak_wrapped_args.push_back(GArg(&(oak_ins.back())));
+        return oak_wrapped_args.back();
+        break;
+    default:
+        util::throw_error(std::logic_error("Unsupported GShape type in OAK backend"));
+        break;
+    }
+}
+
+void cv::gimpl::GOAKExecutable::outArg(const RcDesc &rc,
+                                       std::vector<ExtractTypeHelper::OutputPtr>& oak_outs) {
+    switch (rc.shape) {
+    case GShape::GFRAME:
+        oak_outs.push_back(nullptr);
+        break;
+    case GShape::GARRAY:
+        oak_outs.push_back(nullptr);
```

Comment:
+1

Reply:
Since we unpack vector of inputs and outputs for the kernel API - how do we modify it there otherwise? Here we set them to nullptr to modify in the kernel

Reply:
could you put problem description or final code snippet which caused the probem please?